Binaries/DafnyRuntime.cs

```diff
@@ -853,14 +851,16 @@ internal class ConcatSequence<T> : Sequence<T> {
 
     public override T[] Elements {
       get {
-        if (elmts == null) {
+        // IsDefault returns true if the underlying array is a null reference
+        // https://devblogs.microsoft.com/dotnet/please-welcome-immutablearrayt/
+        if (elmts.IsDefault) {
           elmts = ComputeElements();
           // We don't need the original sequences anymore; let them be
           // garbage-collected
           left = null;
           right = null;
         }
-        return elmts;
+        return elmts.ToArray();
```

Comment:
This must make a copy of the array, I suppose. Where is `Elements` used? Does the compiler generate code like `Elements[i]` that would be better as some `GetElement(i)` method?

Reply:
`Elements` is actually used throughout `public abstract class Sequence<T>`, though in cases where we mutate we actually do proper Array copies instead.

To be safer, I can replace these with Immutable usage though (so we don't need to worry if we accidentally miss any copies).